scripts/biomedical/diseaseOntology/format_disease_ontology.py

```diff
@@ -0,0 +1,293 @@
+    return df
+
+
+def icd10_query(df):
+    """
+    Queries the ICD10 ids present in the dataframe,
+    on datacommons, fetches their dcids and adds
+    it to the same column. 
+    Args:
+        df = dataframe to change
+    Returns
+        df = modified dataframe with ICD dcid added
+    """
+    df_temp = df[df.ICD10CM.notnull()]
+    list_icd10 = "ICD10/" + df_temp['ICD10CM'].astype(str)
+    arr_icd10 = shard(list_icd10, 1000)
+    for i in range(len(arr_icd10)):
+        query_str = """
+        SELECT DISTINCT ?id 
+        WHERE {{
+        ?element typeOf ICD10Code .
+        ?element dcid ?id .
+        ?element dcid {0} .
+        }}
+        """.format(arr_icd10[i])
+    result1 = dc.query(query_str)
+    result1_df = pd.DataFrame(result1)
+    result1_df['element'] = result1_df['?id'].str.split(pat="/").str[1]
+    result1_df.columns = ['ICD10_id', 'element']
+    df = pd.merge(result1_df, df, left_on='element', right_on='ICD10CM', how = "right")
+    return df
+
+
+def remove_newline(df):
+    df.loc[2505, 'IAO_0000115'] = df.loc[2505, 'IAO_0000115'].replace("\\n", "")
+    df.loc[2860, 'IAO_0000115'] = df.loc[2860, 'IAO_0000115'].replace("\\n", "")
+    df.loc[2895, 'IAO_0000115'] = df.loc[2895, 'IAO_0000115'].replace("\\n", "")
+    df.loc[2934, 'IAO_0000115'] = df.loc[2934, 'IAO_0000115'].replace("\\n", "")
+    df.loc[3036, 'IAO_0000115'] = df.loc[3036, 'IAO_0000115'].replace("\\n", "")
+    return df
+
+
+def wrapper_fun(file_input):
+    # Read disease ontology .owl file
+    tree = ElementTree.parse(file_input)
+    # Get file root
+    root = tree.getroot()
+    # Find owl classes elements
+    all_classes = root.findall('{http://www.w3.org/2002/07/owl#}Class')
+    # Parse owl classes to human-readble dictionary format
+    parsed_owl_classes = []
+    for owl_class in all_classes:
+        info = list(owl_class.iter())
+        parsed_owl_classes.append(parse_do_info(info))
+    # Convert to pandas Dataframe
+    df_do = pd.DataFrame(parsed_owl_classes)
+    format_cols(df_do)
+    df_do = df_do.drop([
+        'Class', 'exactMatch', 'deprecated', 'hasRelatedSynonym', 'comment',
+        'OBI_9991118', 'narrowMatch', 'hasBroadSynonym', 'disjointWith',
+        'hasNarrowSynonym', 'broadMatch', 'created_by', 'creation_date',
+        'inSubset', 'hasOBONamespace'
+    ],
+                       axis=1)
+    df_do = col_explode(df_do)
+    df_do = mesh_query(df_do)
+    df_do = icd10_query(df_do)
+    col_string(df_do)
+    df_do = df_do.drop(['A', 'B', 'nan', 'hasDbXref', 'KEGG'], axis=1)
+    df_do = df_do.drop_duplicates(subset='id', keep="last")
+    df_do = df_do.reset_index(drop=True)
+    df_do = df_do.replace('"nan"', np.nan)
+    df_do = df_do.replace("nan", np.nan)
+    #generate dcids
+    df_do['id'] = "bio/" + df_do['id']
+    df_do['subClassOf'] = "bio/" + df_do['subClassOf']
+    df_do['ICD9CM'] = "ICD10/" + df_do['ICD9CM'].apply(str)
+    return df_do
+
+def main():
```

Comment:
Please confirm that this is the most up-to-date version of the script that includes changes like how to handle lists of texts values that have commas within a single cell value and other changes that we've previously discussed.